pkgs/top-level/darwin-packages.nix

```diff
@@ -20,6 +20,18 @@ let
   };
 
   aliases = self: super: lib.optionalAttrs config.allowAliases (import ../top-level/darwin-aliases.nix lib self super pkgs);
+
+  mkBootstrapStdenv =
+    stdenv:
+    stdenv.override (old: {
+      extraBuildInputs = map (
+        pkg:
+        if lib.isDerivation pkg && lib.getName pkg == "apple-sdk" then
+          pkg.override { enableBootstrap = true; }
```

Comment:
`extraBuildInputs` can contain things other than the SDK, so I’m trying to preserve those while doing the replacement. How else would I detect the SDK to swap it with a bootstrap version?

The reason for having a `bootstrapStdenv` is because it’s needed to build packages that are propagated by the SDK. If the SDK that propagates them were used to build them, it would cause an infinite recursion.

Note that this isn’t done via overrides in the stdenv bootstrap because that doesn’t work for cross-compilation.